lib/helpers.js

```diff
@@ -79,15 +97,28 @@ exports.validateRequiredName = function (name) {
  * @returns {String|Boolean}
  */
 exports.validateOptionalName = function (name) {
-  if (name.match(/[\/@\s\+%:\.]/)) {
-    return 'Name cannot contain special characters (/@+%:. ): ' + name;
+  var result = validateValue(name, /[\/@\s\+%:\.]/);
+  if (result !== true) return result;
+  return true;
+};
+
+/**
+ * Validate value and throw error on unallowed characters.
+ * @param {String} name The user input
+ * @param {String} unallowedCharacters Characters that are not allowed.
+ * @returns {String|Boolean}
+ */
+function validateValue(name, unallowedCharacters) {
```

Comment:
The name `validateValue` is not accurate, because it creates impression that any/all values should be validated through this function.

I am proposing to change `validateOptionalName` to accept the new parameter `unallowedCharacters` and call it from `validateRemoteMethodName`. That way we don't need `validateValue` at all.

Thoughts?